.github/workflows/tsan.yaml

```diff
@@ -54,8 +65,18 @@ jobs:
           # Check whether free-threading mode is enabled
           PYTHON_GIL=0 ${GITHUB_WORKSPACE}/cpython-tsan/bin/python3 -c "import sys; assert not sys._is_gil_enabled()"
 
+      - name: Save CPython with TSAN
+        id: cache-cpython-tsan-save
+        uses: actions/cache/save@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
+        with:
+          path: ./cpython-tsan
+          key: ${{ runner.os }}-cpython-tsan-${{ hashFiles('cpython/configure.ac') }}
+
+      - name: Create CPython archive
```

Comment:
Wouldn't it make sense to move the .tgz formation under the cache as well?

Reply:
Cache action makes an archive of the folder to cache. So we can try to ask it to cache the archive, so it will be an archive of an archive...